tiledb/api/src/stats.rs

```diff
@@ -0,0 +1,70 @@
+use crate::error::Error;
+use crate::Result as TileDBResult;
+
+pub fn stats_enable() -> TileDBResult<()> {
+    let c_ret : i32 = unsafe {
+        ffi::tiledb_stats_enable()
+    };
+
+    if c_ret == ffi::TILEDB_OK {
+       Ok(())
+    } else {
+        Err(Error::from("TileDB stats were not successfully enabled."))
+    }
+}
+
+pub fn stats_disable() -> TileDBResult<()> {
+    let c_ret : i32 = unsafe {
+        ffi::tiledb_stats_disable()
+    };
+
+    if c_ret == ffi::TILEDB_OK {
+        Ok(())
+    } else {
+        Err(Error::from("TileDB stats were not successfully disabled."))
+    }
+}
+
+pub fn stats_reset() -> TileDBResult<()> {
+    let c_ret : i32 = unsafe {
+        ffi::tiledb_stats_reset()
+    };
+
+    if c_ret == ffi::TILEDB_OK {
+       Ok(())
+    } else {
+       Err(Error::from("TileDB stats were not successfully reset."))
+    }
+}
+
+pub fn stats_dump_str() -> TileDBResult<String> {
+    let mut stats_dump_c_str = std::ptr::null_mut::<std::os::raw::c_char>();
+    let res = unsafe {
+        ffi::tiledb_stats_dump_str(
+            &mut stats_dump_c_str as *mut *mut std::os::raw::c_char,
+        )
+    };
+    if res == ffi::TILEDB_OK {
+        assert!(!stats_dump_c_str.is_null());
+        let stats_dump = unsafe { std::ffi::CStr::from_ptr(stats_dump_c_str) };
```

Comment:
I'm pretty sure we're leaking the string returned by core here. I think an easy way to clean this up would be to add a RawStatsString for the `*mut std::ffi::c_char` pointer that implements drop, and then we can probably wrap the `RawStatsString` -> `String` conversion using a TryFrom or something.

Reply:
Isn't there `tiledb::string::String` for this purpose?

Reply:
When I implemented this I assumed that the memory allocated from tiledb_stats_dump_str would be moved into the Rust string object and thus not need to be memory managed further. Although yeah it would be nice to implement a string wrapper type that drops as needed. I think I'll just implement a wrapper class that implements drop?

I think tiledb_string_t is a wrapper around the tiledb_string_t class, not necessarily for handling C to Rust string conversions. But I'm probably 75% sure of that.